frontend/src/hooks/useNavigation.ts

```diff
@@ -39,20 +39,15 @@ export function useNavigation(): INavigationHook {
     remoteUI: isRemoteUI(state),
   }))
   const [shouldUpdate, setShouldUpdate] = useState<boolean>(true)
+  const [navigationPath, setNavigationPath] = useState({})
 
   const match = location.pathname.match(REGEX_FIRST_PATH)
   const menu = match ? match[0] : '/devices'
 
-  // Used for remembering the state of a tab - no longer used
-  const recallPath = (selected: string) => {
-    const stored = navigation[selected]
-    if (!stored || stored === location.pathname) return selected
-    else return stored
-  }
-
   useEffect(() => {
     if (navigation[menu] !== location.pathname) {
-      ui.set({ navigation: { ...navigation, [menu]: location.pathname } })
+      ui.set({ ...navigationPath, navigation: { ...navigation, [menu]: location.pathname } })
```

Comment:
When you enter navigationPath is empty and if that variable is changed it will not be update.
This is a question, shall we need to add it on the final array of useEffect?

Reply:
navigationPath is an object and it is using updated key to override the reducer